src/modules/claims/claims.service.ts

```diff
@@ -261,28 +262,61 @@ export class ClaimsService {
     return this._cacheClient.issueClaim(this._signerService.did, message);
   }
 
+  /**
+   *
+   * @param token optional token containing claimType, version and subject
+   * @returns claim params obtained from token
+   */
+  private getClaimTypeFromToken(token?: string) {
```

Comment:
token is not ment to be required, user can registerOnchain with claim data lacking of token.

Reply:
`token` is optional because this method is called when registering with and without request. I think that this method should be called only when registering with request